src/main/java/ch/heigvd/amt/stack/infrastructure/persistence/remote/RemoteGamificationRepository.java

```diff
@@ -51,25 +124,42 @@ public void postEvent(CredentialId user, GamificationEvent event) {
     }
   }
 
-  public void addCategory(String name, String title, String description) {
-    Category newCategory = new Category()
-        .name(name).title(title).description(description);
+  public void addCategory(Category category) {
     try {
-      categoriesApi.putCategory(name, newCategory);
+      categoriesApi.putCategory(category.getName(), category);
     } catch (ApiException e) {
       e.printStackTrace();
     }
   }
 
-  public void addRule(String name, String categoryName, GamificationEvent eventType,
-      Integer points) {
-    Rule newRule = new Rule()
-        .name(name).category(categoryName)
-        .event(eventType.name()).points(points);
+  public void addRule(Rule newRule) {
     try {
       rulesApi.postRule(newRule);
     } catch (ApiException e) {
       e.printStackTrace();
     }
   }
+
+  public void addBadge(Badge newBadge) {
+    try {
+      badgesApi.putBadge(newBadge.getName(), newBadge);
+    } catch (ApiException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public List<Badge> getBadges(CredentialId user) {
+    List<Badge> result = new java.util.ArrayList<>(List.of());
+    try {
+      var agg = aggregatesApi.getUserAggregate(user.toString(),
+          categories.stream().map(Category::getName).collect(Collectors.toList()));
```

Comment:
Could we reformat ? I find it clearer

```suggestion
      var agg = aggregatesApi.getUserAggregate(
          user.toString(),
          categories.stream().map(Category::getName).collect(Collectors.toList())
      );
```